### PR TITLE
Release sql microsoft.sql 2024 05 01 preview

### PR DESCRIPTION
**Pull request by ravigautam1896**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by dhruvesh** [comment]
This PR has at least one breaking change (label: BreakingChangeReviewRequired).
To unblock this PR, follow the process at [aka.ms/brch](https://aka.ms/brch).

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (dhruvesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:31]: can we reuse this across instancePools.json and intancePoolOperation.json. We are duplicating this

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (dhruvesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:86]: Its redundant

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (dhruvesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:215]: errorcode should come on 4xx or 5xx status code, we should create another definition, and explicity link to 5xx status code response

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/ManagedInstanceLongTermRetentionPolicies.json
Comment (dhruvesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/ManagedInstanceLongTermRetentionPolicies.json:247]: why are we adding locations in header, can we add relevant example.

Comment (dhruvesh) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/ManagedInstanceLongTermRetentionPolicies.json:247]: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers

Please refer this for guidelines

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (dhruvesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:144]: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten

Comment (dhruvesh) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:144]: Not reccmmedndedd to use

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (dhruvesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:198]: Consider using provisioningState https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property

**Comment by dhruvesh** [comment]
ARMSignOff already done

PR #1
[Update for Import/Export with Managed Identity by psong-ms · Pull Request #29596 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/29596)
 
PR #2
[Set the additionalProperties type to string for DatabaseAdvisors API by UrosKrsticMIPerf · Pull Request #29823 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/29823)
 
PR #3
[Dev/ivarandj/instance pool operations api by ivanarandjelovic · Pull Request #29932 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/29932)
 
PR #4
[Adding updated DatabaseOperations.json file by akromm · Pull Request #29956 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/29956)
 
PR #5
[Adding swagger files for minor changes related to release-sql-Microsoft.Sql-2024-05-01-preview by ravigautam1896 · Pull Request #30030 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/30030)

**Comment by dhruvesh** [comment]
LintDiff are not reflecting, seems like LintDiff Automation is broken in this PR

**Comment by dhruvesh** [comment]
https://github.com/Azure/azure-rest-api-specs/pull/29956#issuecomment-2250716281

Model Validation is approved here

**Review by ericshape — APPROVED** [review]

**Comment by dhruvesh** [comment]
image

68 Lintdiff are showing, but those lintdiff are not because of recent changes. Wanted to understand more does Lintdiff check incrementat changes only, or does it scan all the files

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseAdvisors.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseAdvisors.json:409]: > "additionalProperties": true, [](http://example.com/codeflow?start=9&length=30)

what is this change referring to ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json:36]: refer from common types v6, if not v5

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json:320]: if you are not using, do you need these definitions ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json:45]: > "pattern": "^[a-z0-9]+(-*[a-z0-9]+)*$" [](http://example.com/codeflow?start=11&length=39)

if you are adding patterns now, will the users who have serverNames outside this pattern, cannot use the new apiversion then ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json:55]: >"pattern": "^[^<>*%&:\\\\\\/?\\x00-\\x1F]*[^.<>*%&:\\\\\\/?\\s\\x00-\\x1F]$" [](http://example.com/codeflow?start=12&length=76)

if you are adding patterns now, will the users who have databaseName outside this pattern, cannot use the new apiversion then ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json:320]: these are also defined in DatabaseRecommendedActions.json. Could you have a single file for these shared types and refer from that ?


This is also the same pattern for other defintions too - they are repeated in each swagger file. Its better to have a common place to define them and then refer from there.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:573]: patterns are added in DatabaseOperations.json's /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/operations .

Why the inconsistency ?

**Comment by ramoka178** [comment]
>               }

add description

---
Refers to: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DistributedAvailabilityGroups.json:428 in 8e3d118. [](commit_id = 8e3d1188a0a09b93a18ba84d78164a44d846e3b0, deletion_comment = False)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:107]: v5/v6

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:28]: >./common/v1/types.json#/parameters/ResourceGroupParamete [](http://example.com/codeflow?start=28&length=56)

use latest common-types

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:261]: >Gets the name of the impact dimension. e.g., CPUChange, DiskSpaceChange, NumberOfQueriesAffected [](http://example.com/codeflow?start=26&length=96)

fix the description

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:45]: use from common-types latest

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:74]: >"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/instancePools/{instancePoolName}/operations/{operationId} [](http://example.com/codeflow?start=4&length=149)

what about the Create/Update/Delete of this operation? how are they orchestrated ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:83]: >v1 [](http://example.com/codeflow?start=37&length=2)

v5/v6

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:498]: > "additionalProperties": true, [](http://example.com/codeflow?start=9&length=30)

what does this imply to have as true ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:137]: >v1 [](http://example.com/codeflow?start=35&length=2)

v5/v6

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:144]: is this still needed and why ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:181]: >operationFriendlyName [](http://example.com/codeflow?start=9&length=21)

displayName ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:186]: >percentComplete [](http://example.com/codeflow?start=9&length=15)

operationCompletedPercentage?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:240]: why are these needed in properties ? shouldn't they online come in error responses. and there is an error response definition in common-types

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:254]: >boolean [](http://example.com/codeflow?start=19&length=7)

ARM recommends enums over booleans for future proof APIs. 

 

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible. 

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its cofusing, and no better than a boolean.

**Comment by ramoka178** [comment]
>                 "type": "string"

update description

---
Refers to: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/ManagedInstanceLongTermRetentionPolicies.json:249 in 8e3d118. [](commit_id = 8e3d1188a0a09b93a18ba84d78164a44d846e3b0, deletion_comment = False)

**Comment by ramoka178** [comment]
>           "additionalProperties": true,

what does this change imply ?

---
Refers to: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/ServerAdvisors.json:586 in 8e3d118. [](commit_id = 8e3d1188a0a09b93a18ba84d78164a44d846e3b0, deletion_comment = False)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ramoka178) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:198]: yeah, why not use existing definition ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:74]: Instance pool operations cannot be create, updated and deleted via API. They are implicitly created by the control plan when crate/update/delete is executed on the instance pool it self.

This is why we have only list and get for instance pool operations. Same patter is used for SQL Managed Instance (and SQL DB, I believe).

Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:74]: btw, this was already answered in the PR introducing this change.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:107]: SQL is not onboarded on V5/v6, already answered in original PR.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:83]: SQL not onboarded to v5/v6

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:137]: same as above

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:181]: Keeping in sync with SQL Managed Instance operations

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:186]: Same as for Managed instance, keeping request/response objects as similar as possible

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json
Comment (ivanarandjelovic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json:254]: Agree, I already changed errorType from boolean to enum. Can you suggest how to convert this to enum? 

It's a bit of a struggle to come up with property name that would have values "Cancelable", "NotCancelable". Maybe "CancelationType"?

As this is also same as in SQL MI apis, and since Portal work is already done maybe we should leave it "as is" for now and change it in next API release. Would this be acceptable?

**Comment by ravigautam1896** [comment]
@ramoka178  in this PR we are just merging the release branch to main branch, no new changes are there in this PR.
and the changes which you have mentioned.
these are already approved.

ARMSignOff already done

PR https://github.com/Azure/azure-rest-api-specs/pull/1
https://github.com/Azure/azure-rest-api-specs/pull/29596

PR https://github.com/Azure/azure-rest-api-specs/pull/2
https://github.com/Azure/azure-rest-api-specs/pull/29823

PR https://github.com/Azure/azure-rest-api-specs/pull/3
https://github.com/Azure/azure-rest-api-specs/pull/29932

PR https://github.com/Azure/azure-rest-api-specs/pull/4
https://github.com/Azure/azure-rest-api-specs/pull/29956

PR https://github.com/Azure/azure-rest-api-specs/pull/5
https://github.com/Azure/azure-rest-api-specs/pull/30030

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json:55]: I assume the pattern is chosen for backwards compatibility

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:573]: Yes, why doesn't the regex pattern allow the example property value to be 'Default-SQL-SouthEastAsia'?

>String does not match pattern ^[\p{L}\p{N}().-]*[\p{L}\p{N}()-]$:

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:573]: It actually seems to be the ResourceGroupName regex which is the problem?

>        "name": "resourceGroupName",
            "in": "path",
            "description": "The name of the resource group that contains the resource.",
            "required": true,
            "type": "string",
            "maxLength": 90,
            "minLength": 1,
            "pattern": "^[\\p{L}\\p{N}_().-]*[\\p{L}\\p{N}_()-]$"

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:573]: I think it would be better if you were using a reference ($ref) to common types definition of ResourceGroupParameter 

https://github.com/Azure/azure-rest-api-specs/edit/main/specification/common-types/resource-management/v5/types.json

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:573]: i.e. the regex isn't correct vs the spec:

https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftresources

**Comment by TimLovellSmith** [comment]
> @ramoka178 in this PR we are just merging the release branch to main branch, no new changes are there in this PR. and the changes which you have mentioned. these are already approved.
> 
> ARMSignOff already done
> 
> PR #1 #29596
> 
> PR #2 #29823
> 
> PR #3 #29932
> 
> PR #4 #29956
> 
> PR #5 #30030

You still need to fix validation errors before merging this PR. For the resource group name parameter the fix should be easy just reference common-types v5 or v6 ResourceGroupParameter the way other RPs do

File: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseRecommendedActions.json:201]: >RecommendedAction [](http://example.com/codeflow?start=5&length=17)

this is hard to review the change because its been reordered

**Comment by TimLovellSmith** [comment]
>           "readOnly": true

why change it?

---
Refers to: specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/LocationCapabilities.json:1654 in 8a9361a. [](commit_id = 8a9361aae4351dd00446f5644f94869c15660a2e, deletion_comment = True)

**Comment by TimLovellSmith** [comment]
Please fix model validation errors

**Pull request closed without merging**
